generator/sbpg/targets/resources/sbp_messages_mod.rs

```diff
@@ -22,21 +22,22 @@ use self::(((p.identifier|mod_name)))::(((m.identifier|camel_case)));
 
 #[cfg(feature = "serialize")]
 use serde::{Serialize, Deserialize};
+use crate::serialize::SbpSerialize;
 
 pub trait SBPMessage {
-    const MSG_ID: u16;
-
+    fn get_message_type(&self) -> u16;
     fn get_sender_id(&self) -> Option<u16>;
     fn set_sender_id(&mut self, new_id: u16);
+    fn into_frame(self) -> Vec<u8>;
```

Comment:
I ended up just taking a `&dyn SBPMessage` to the framing function and making the `SBPMessage` inherit/extend the `SbpSerialize` trait.

Reply:
This is great.

Just for fun I played around a bit to see if we could bring back the `SBPMessage::to_frame` trait method for ergonomics' sake. It's basically the same as before in terms of code duplication since it can't be a default trait implementation due to the `Sized` limitation you mentioned before, but at least it's effectively the same implementation with dispatches to `sbp::framer::to_frame` instead of having different generated implementations. Curious what you guys think.

https://github.com/swift-nav/libsbp/pull/758